gr-filter/lib/firdes.cc

```diff
@@ -571,6 +571,7 @@ vector<float> firdes::root_raised_cosine(
         } else {
             if (alpha == 1) {
                 taps[i] = -1;
+                scale += taps[i];
```

Comment:
This line is already below outside the `if` statement ... as well as setting `taps[i]` ... I think the duplication is problematic. I know it was here before this PR ... but, can you please verify the correct algorithm & fix this up?

Reply:
There is a `continue` on the next line, so the duplicate lines you refer to are not executed when the condition of this if statement is true.

I'm not familiar with whatever manipulations were performed to the canonical root raised cosine formula to produce the form used in the `else` part of this. I started attempting to understand the approach in this function by manipulation of the standard formula I'm familiar with but did not get far in the time I have available presently.

Reply:
Ah ... yes, yes there is a `continue` ... somehow missed that ... my bad! Thanks for pointing it out to me. Let me re-review with that new info in mind!